From 557c5a8583429f5ec90a50dd16785df459c7697e Mon Sep 17 00:00:00 2001
From: Rubidium <rubidium@openttd.org>
Date: Sun, 15 Jan 2023 08:51:00 +0100
Subject: [PATCH 0841/1053] Fix #10057: FallbackParagraphLayout fails to
 properly wrap

... during the first word after a new run has been started.
---
 src/gfx_layout.cpp | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/src/gfx_layout.cpp b/src/gfx_layout.cpp
index 40f9adeb94..1a163330d2 100644
--- a/src/gfx_layout.cpp
+++ b/src/gfx_layout.cpp
@@ -552,8 +552,6 @@ std::unique_ptr<const ParagraphLayouter::Line> FallbackParagraphLayout::NextLine
 
 			next_run = this->buffer_begin + iter->first;
 			begin = this->buffer;
-
-			last_space = nullptr;
 		}
 
 		if (IsWhitespace(c)) last_space = this->buffer;
@@ -591,7 +589,7 @@ std::unique_ptr<const ParagraphLayouter::Line> FallbackParagraphLayout::NextLine
 		this->buffer++;
 	}
 
-	if (l->size() == 0 || last_char - begin != 0) {
+	if (l->size() == 0 || last_char - begin > 0) {
 		int w = l->GetWidth();
 		l->emplace_back(iter->second, begin, last_char - begin, w);
 	}
-- 
2.55.0

